File: sdc-workflow-designer-ui/src/app/model/workflow/timer-event-definition.ts
/**
 * Copyright (c) 2017 ZTE Corporation.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * and the Apache License 2.0 which both accompany this distribution,
 * and are available at http://www.eclipse.org/legal/epl-v10.html
 * and http://www.apache.org/licenses/LICENSE-2.0
 *
 * Contributors:
 *     ZTE - initial API and implementation and/or initial documentation
 */
import { WorkflowNode } from './workflow-node';

export enum TimerEventDefinitionType {
    timeDate,
    timeDuration,
    timeCycle,
}

export class TimerEventDefinition {
    constructor(public type: string, // 'timeDate',  'timeCycle', 'timeDuration'
        public timeDate?: string, // <timeDate>10/10/2099 00:00:00</timeDate>
        public timeDuration?: string, // ISO 8601  P1Y3M5DT6H7M30S
        public timeCycle?: string) { // ISO 8601  P1Y3M5DT6H7M30S

    }
}

File: sdc-workflow-designer-server/src/main/java/org/onap/sdc/workflowdesigner/model/TimerEventDefinition.java
/**
 * Copyright (c) 2017 ZTE Corporation.
 * All rights reserved. This program and the accompanying materials
 * are made available under the Apache License, Version 2.0
 * and the Eclipse Public License v1.0 which both accompany this distribution,
 * and are available at http://www.eclipse.org/legal/epl-v10.html
 * and http://www.apache.org/licenses/LICENSE-2.0
 *
 * Contributors:
 *     ZTE - initial API and implementation and/or initial documentation
 */
package org.onap.sdc.workflowdesigner.model;

public class TimerEventDefinition {
    private String type;
    private String timeDuration;
    private String timeDate;
    private String timeCycle;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getTimeDuration() {
        return timeDuration;
    }

    public void setTimeDuration(String timeDuration) {
        this.timeDuration = timeDuration;
    }

    public String getTimeDate() {
        return timeDate;
    }

    public void setTimeDate(String timeDate) {
        this.timeDate = timeDate;
    }

    public String getTimeCycle() {
        return timeCycle;
    }

    public void setTimeCycle(String timeCycle) {
        this.timeCycle = timeCycle;
    }

}
